chain/client/src/sync.rs

```diff
@@ -557,12 +555,17 @@ impl BlockSync {
         };
 
         if let Some(peer) = peer {
+            debug!(target: "sync", "Block sync: {}/{} requesting block {} from {} (out of {} peers)",
+		   head.height, header_head.height, next_hash, peer.peer_info.id, highest_height_peers.len());
             self.network_adapter.do_send(PeerManagerMessageRequest::NetworkRequests(
                 NetworkRequests::BlockRequest {
                     hash: request.hash,
                     peer_id: peer.peer_info.id.clone(),
                 },
             ));
+        } else {
+            warn!(target: "sync", "Block sync: {}/{} No available {}peers to request block {} from",
```

Comment:
(in a separate PR) - we should consider having a metric with this (exported to grafana) - to see often this happens.

Reply:
Yeah would be good to have stats on that, agreed